Mathlib/LinearAlgebra/Matrix/DotProduct.lean

```diff
@@ -74,30 +72,57 @@ end Semiring
 
 section Self
 
-variable [Fintype n]
+variable  [Fintype m] [Fintype n] [Fintype p]
 
 @[simp]
 theorem dotProduct_self_eq_zero [LinearOrderedRing R] {v : n → R} : dotProduct v v = 0 ↔ v = 0 :=
   (Finset.sum_eq_zero_iff_of_nonneg fun i _ => mul_self_nonneg (v i)).trans <| by
     simp [Function.funext_iff]
 #align matrix.dot_product_self_eq_zero Matrix.dotProduct_self_eq_zero
 
+section StarOrderedRing
+
+variable [PartialOrder R] [NonUnitalRing R] [StarOrderedRing R] [NoZeroDivisors R]
+
 /-- Note that this applies to `ℂ` via `Complex.strictOrderedCommRing`. -/
 @[simp]
-theorem dotProduct_star_self_eq_zero [PartialOrder R] [NonUnitalRing R] [StarOrderedRing R]
-    [NoZeroDivisors R] {v : n → R} : dotProduct (star v) v = 0 ↔ v = 0 :=
+theorem dotProduct_star_self_eq_zero {v : n → R} : dotProduct (star v) v = 0 ↔ v = 0 :=
   (Finset.sum_eq_zero_iff_of_nonneg fun i _ => (@star_mul_self_nonneg _ _ _ _ (v i) : _)).trans <|
     by simp [Function.funext_iff, mul_eq_zero]
 #align matrix.dot_product_star_self_eq_zero Matrix.dotProduct_star_self_eq_zero
 
 /-- Note that this applies to `ℂ` via `Complex.strictOrderedCommRing`. -/
 @[simp]
-theorem dotProduct_self_star_eq_zero [PartialOrder R] [NonUnitalRing R] [StarOrderedRing R]
-    [NoZeroDivisors R] {v : n → R} : dotProduct v (star v) = 0 ↔ v = 0 :=
+theorem dotProduct_self_star_eq_zero {v : n → R} : dotProduct v (star v) = 0 ↔ v = 0 :=
   (Finset.sum_eq_zero_iff_of_nonneg fun i _ => (@star_mul_self_nonneg' _ _ _ _ (v i) : _)).trans <|
     by simp [Function.funext_iff, mul_eq_zero]
 #align matrix.dot_product_self_star_eq_zero Matrix.dotProduct_self_star_eq_zero
 
+@[simp]
+lemma conjTranspose_mul_self_eq_zero_iff {A : Matrix m n R} : (Aᴴ⬝A) = 0 ↔ A = 0 :=
+  ⟨fun h => Matrix.ext fun i j =>
+    (congr_fun <| dotProduct_star_self_eq_zero.1 <| Matrix.ext_iff.2 h j j) i,
+  fun h => h ▸ Matrix.mul_zero _⟩
+
+@[simp]
+lemma self_mul_conjTranspose_eq_zero_iff {A : Matrix m n R} : (A⬝Aᴴ) = 0 ↔ A = 0 :=
+  ⟨fun h => Matrix.ext fun i j =>
+    (congr_fun <| dotProduct_self_star_eq_zero.1 <| Matrix.ext_iff.2 h i i) j,
+  fun h => h ▸ Matrix.zero_mul _⟩
+
+lemma ker_conj_transpose_mul_self_eq_ker (A: Matrix m n R) (B: Matrix n p R) :
```

Comment:
I think `ker` is confusing here, because a) we don't have a definition of `Matrix.ker`, and b) if we did it would be a set / submodule, not proposition
```suggestion
lemma conj_transpose_mul_self_mul_eq_zero_iff (A: Matrix m n R) (B: Matrix n p R) :
```